flocker/node/_deploy.py

```diff
@@ -735,5 +768,8 @@ def find_dataset_changes(hostname, current_state, desired_state):
         local_current_dataset_ids | remote_current_dataset_ids)
     creating = set(dataset for dataset in local_desired_datasets
                    if dataset.dataset_id in creating_dataset_ids)
+
+    # XXX all datasets with deleted=True attribute should be added to deleted
+    # set on the DatasetChanges
```

Comment:
`find_dataset_changes` is getting somewhat long.  Any way we can factor any of those code into separate helpers?